xscen/aggregate.py

```diff
@@ -467,3 +488,293 @@ def spatial_mean(
+    Compute the climatology of indicators for a given warming level.
+
+    First, this function calculates the start and end year of a window when the given
+    global warming level is reached from a csv of yearly global mean temperature. Second,
+    the function computes indicators and interrannual mean on this same window. Third,
+    it unstacks the season and month to different dimensions and adds a dimension 'horizon' for the warming level.
+    The 'horizon' dimension is added in order to facilicate the concatenation with a
+    reference climatological mean before calculating the deltas.
+
+    Parameters
+    ----------
+    ds: xr.Dataset
+      Input dataset.
+      The dataset should include attributes to help recognize it and find its warming levels - 'cat:mip_era', 'cat:experiment', 'cat:member', and either 'cat:source' for global models or 'cat:driving_model' for regional models.
+    wl: float
+      Warming level.
+      eg. 2 for a global warming level of +2 degree Celsius above the mean temperature of the `tas_ref_period`.
+    indicators:  Union[str, PosixPath, Sequence[Indicator], Sequence[Tuple[str, Indicator]]]
+      Indicators to compute. It will be passed to the `indicators` argument of `xs.compute_indicators`.
+    window: int
+      Size of the window over which to compute both the warming levels and the
+      climatological mean of the indicators. The rolling operation is centered.
+    min_periods: int
+      Minimum number of years required for climatological mean to be computed.
+      If left at None, it will be window-2. This is to get a non-nan value for periods
+      that go from december to january (ex. DJF in QS-DEC indicators) because they will have
+      less occurrences than the window size because nans are dropped.
+    tas_baseline_period: list
+      Base period. The warming is calculated with respect to this period. The default is ["1850", "1900"].
+    ignore_member: bool
+      Whether to ignore the member when searching for the model run in tas_csv.
+    tas_csv: str
+      Path to a csv of annual global mean temperature.
+      If None, it will default to data/IPCC_annual_global_tas.csv which was built from
+      the IPCC atlas data from  Iturbide et al., 2020 (https://doi.org/10.5194/essd-12-2959-2020)
+      and extra data from pilot models of MRCC5 and ClimEx.
+    to_level:
+      The processing level to assign to the output.
+      Use "{wl}" in the string to dynamically include the warming level.
+
+    Returns
+    -------
+    xr.Dataset
+        Warming level dataset
+
+    """
+    if min_periods is None:
+        min_periods = window - 2
+
+    if tas_baseline_period is None:
+        tas_baseline_period = ["1850", "1900"]
+
+    if tas_csv is None:
+        tas_csv = Path(__file__).parent / "data/IPCC_annual_global_tas.csv"
+
+    # get info on ds
+    id_ds = ds.attrs["cat:id"]
+    source_ds = (
+        ds.attrs["cat:source"]
+        if "cat:driving_model" not in ds.attrs
```

Comment:
```suggestion
        if not pd.isna(ds.attrs.get("cat:driving_model", None))
```
I think? We also want to catch cases where this is NaN or None.

Reply:
sur mes trucs CMIP6 en moment, `cat:driving_model` n'existe pas, mais si tu dis que tu l'as vu on peut faire ta suggestion. Je ne pense pas que ça fait du mal.

Reply:
J'essaie de me rappeler où (pas les Notebooks) et c'est peut-être un problème qu'on avait et qui est réglé maintenant, mais en effet ça ne coûte pas cher d'ajouter cette vérification.